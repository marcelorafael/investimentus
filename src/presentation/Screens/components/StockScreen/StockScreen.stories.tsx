import { action } from '@storybook/addon-actions';
import { text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import React from 'react';
import { StockScreen } from './StockScreen'

storiesOf('Screens/StockScreen', module)
  .addDecorator(withKnobs)
  .add('default', () => (
    <StockScreen />
  ))
