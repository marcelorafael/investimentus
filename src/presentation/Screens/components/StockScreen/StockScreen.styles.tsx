import styled from 'styled-components/native'
import LinearGradient from 'react-native-linear-gradient'

// background-color: ${props => props?.theme?.COLORS?.greenFocused ?? ''};

export const Container = styled(LinearGradient).attrs({
  colors: ['#000', '#7F5DF0']
})`
  flex: 1;
  justify-content: space-between;
`

// background-color: ${({theme}) => theme?.COLORS?.white ?? '#fff'};

// colors:['#accec6','#1a3f1a']
export const Footer = styled(LinearGradient).attrs({
  colors: ['#000', '#7F5DF0']
})`
  width: 100%;
  height: 15%;
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;

  justify-content: center;
`

export const DateFormat = styled.Text`
  ${({ theme }) => theme?.FONTS?.bullpen ?? {}}
  text-align: center;
  color: ${({ theme }) => theme?.COLORS?.white ?? '#FFF'};
`

export const TextFooter = styled.Text`
  ${({ theme }) => theme?.FONTS?.robotoThin ?? {}}
  text-align: center;
  font-size: 14px;
  font-weight: bold;
  color: ${({ theme }) => theme?.COLORS?.white ?? '#FFF'};
`

export const Header = styled.View`
  flex-direction: row;
  width: 100%;
  align-items: center;
  justify-content: center;
  margin-top: 8px;
`

export const Input = styled.TextInput`
  width: 80%;
  height: 50px;

  border-bottom-left-radius: 5px;
  border-top-left-radius: 5px;

  background-color: ${({ theme }) => theme?.COLORS?.white ?? '#fff'};
`

export const BtnSearch = styled.TouchableOpacity.attrs({
  activeOpacity: 0.7
})`
  background-color: ${({ theme }) => theme?.COLORS?.primary ?? '#fff'};
  width: 50px;
  height: 50px;

  align-items: center;
  justify-content: center;

  border-bottom-right-radius: 5px;
  border-top-right-radius: 5px;
`

export const CenterView = styled.View`
  align-items: center;
`
