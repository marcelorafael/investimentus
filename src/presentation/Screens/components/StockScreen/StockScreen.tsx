import React, { useState } from 'react'

import { format } from 'date-fns'
import Icon from 'react-native-vector-icons/FontAwesome'
import {
  Container,
  Footer,
  DateFormat,
  TextFooter,
  Input,
  Header,
  BtnSearch,
  CenterView,
} from './StockScreen.styles';

const date = format(new Date(), 'dd/MM/yyyy');

interface StockScreenProps {
  value: string
  children: React.ReactElement
  placeholder: string
  onChangeText: () => void
  onPress: () => void
}

const StockScreen: React.FC<StockScreenProps> = (
  props: StockScreenProps
) => {
  const { value, placeholder, children, onChangeText, onPress } = props;


  return (
    <Container>
      <Header>
        <Input
          value={value}
          onChangeText={onChangeText}
          placeholder={placeholder}
        />
        <BtnSearch onPress={onPress}>
          <Icon name="search" color="#FFF" size={20} />
        </BtnSearch>
      </Header>

      <CenterView>
        {children}
      </CenterView>

      <Footer>
        <DateFormat>{date}</DateFormat>
        <TextFooter>Saíba tudo sobre uma determina ação aqui.</TextFooter>
      </Footer>
    </Container>
  )
}
const StockScreenMemo = React.memo(StockScreen);
export { StockScreenMemo as StockScreen }