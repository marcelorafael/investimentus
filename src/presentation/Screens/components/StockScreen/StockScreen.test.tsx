import React from 'react'
import renderer from 'react-test-renderer'
import { StockScreen } from './StockScreen'

describe('Screens/StockScreen', () => {
  it('should match snapshot', () => {
    const sut = (
      <StockScreen
        onChangeText={() => {}}
        onPress={() => {}}
        placeholder=""
        value=""
      >
        <></>
      </StockScreen>
    )

    const renderComponent = renderer.create(sut)

    expect(renderComponent).toMatchSnapshot()
  })
})
