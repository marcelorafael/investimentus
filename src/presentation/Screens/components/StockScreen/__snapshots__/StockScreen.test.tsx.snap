// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screens/StockScreen should match snapshot 1`] = `
<BVLinearGradient
  colors={
    Array [
      4278190080,
      4286537200,
    ]
  }
  endPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "space-between",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 8,
          "width": "100%",
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder=""
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderBottomLeftRadius": 5,
            "borderTopLeftRadius": 5,
            "height": 50,
            "width": "80%",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomRightRadius": 5,
          "borderTopRightRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": 50,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#FFF",
              "fontSize": 20,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
      ]
    }
  />
  <BVLinearGradient
    colors={
      Array [
        4278190080,
        4286537200,
      ]
    }
    endPoint={
      Object {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      Array [
        Object {
          "borderTopLeftRadius": 10,
          "borderTopRightRadius": 10,
          "height": "15%",
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
            "textAlign": "center",
          },
        ]
      }
    >
      09/08/2021
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontSize": 14,
            "fontWeight": "bold",
            "textAlign": "center",
          },
        ]
      }
    >
      Saíba tudo sobre uma determina ação aqui.
    </Text>
  </BVLinearGradient>
</BVLinearGradient>
`
